syntax = "proto3";

import "mlagents_envs/communicator_objects/observation.proto";

option csharp_namespace = "Unity.MLAgents.CommunicatorObjects";
package communicator_objects;

message AgentInfoProto {
    reserved 1; // deprecated repeated float stacked_vector_observation = 1;
    reserved 2; // deprecated repeated bytes visual_observations = 2;
    reserved 3; // deprecated string text_observation = 3;
    reserved 4; // repeated float stored_vector_actions = 4;
    reserved 5; // deprecated string stored_text_actions = 5;
    reserved 6; //repeated float memories = 6;
    float reward = 7;
    bool done = 8;
    bool max_step_reached = 9;
    int32 id = 10;
    repeated bool action_mask = 11;
    reserved 12; // deprecated CustomObservationProto custom_observation = 12;
    repeated ObservationProto observations = 13;
}
